File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(std::vector<int>& nums) {
        std::unordered_set<int> appeared;
        for (int num : nums) {
            if (appeared.find(num) != appeared.end()) {
                return true;
            } else {
                appeared.insert(num);
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/majority-element/submission-2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        threshold = len(nums) // 2

        if threshold <= 1:
            return nums[0]

        hashmap = defaultdict(int)

        for num in nums:
            hashmap[num] += 1

            if hashmap[num] > threshold:
                return num